packages/client/src/pinnedBlocks.ts

```diff
@@ -0,0 +1,52 @@
+import { BlockInfo, PinnedBlocks } from "@polkadot-api/observable-client"
+import { filter, map, mergeMap, Observable, scan, share } from "rxjs"
+import { shareLatest } from "./utils"
+
+export const getNewBlocks$ = (pinnedBlocks$: Observable<PinnedBlocks>) =>
+  pinnedBlocks$.pipe(
+    filter((pinnedBlocks) => !pinnedBlocks.recovering),
+    scan(
+      (acc, pinnedBlocks) => {
+        const newReportedBlocks = new Set<string>()
+        const newBlocks: BlockInfo[] = []
+
+        pinnedBlocks.blocks.forEach(({ hash, number, parent }) => {
+          newReportedBlocks.add(hash)
+          if (!acc.reportedBlocks.has(hash)) {
+            newBlocks.push({
+              hash,
+              number,
+              parent,
+            })
+          }
+        })
+
+        return { reportedBlocks: newReportedBlocks, newBlocks }
+      },
+      {
+        reportedBlocks: new Set<string>(),
+        newBlocks: new Array<BlockInfo>(0),
+      },
+    ),
+    mergeMap(({ newBlocks }) => newBlocks),
+    share(),
+  )
```

Comment:
Maybe we can use the `getPinnedBlocks` observable. It'll emit fewer times, doing essentially less work here.

Reply:
getPinnedBlocks does a bit more though, and it wouldn't be optimal to have it active if it's not needed for anything else.

I can easily add a filter lastEvt == initialized or newBlock though.

Reply:
Yep, that's indeed interesting👍